lib/streamlit/runtime/state/common.py

```diff
@@ -257,7 +305,9 @@ def compute_element_id(
     for k, v in kwargs.items():
         h.update(str(k).encode("utf-8"))
         h.update(str(v).encode("utf-8"))
-    return f"{GENERATED_ELEMENT_ID_PREFIX}-{h.hexdigest()}-{user_key}"
+    element_id = f"{GENERATED_ELEMENT_ID_PREFIX}-{h.hexdigest()}-{user_key}"
+    _register_element_id(element_type, element_id)
```

Comment:
hm my first reaction is that I don't like enhancing the purpose of `compute_element_id` with a complex operation (requires Streamlit internals such as the context). `compute_element_id` is a nice, pure function with a single purpose which is great as it is easily testable etc as it does not require any mocking or setup. If you want to have a convenience function because we often call them together, I would suggest to have a new function `compute_and_register_element_id` that calls the two functions instead of changing the scope.

Reply:
> If you want to have a convenience function because we often call them together

It's less for convenience and more for ensuring that it's actually registered whenever this is used. It's easy to forget the registering if we add new elements or new functionalities requiring IDs. My first version had it separated, but I changed it that way.

I renamed it to `compute_and_register_element_id`

Reply:
I moved it to element utils, which resolves the circular import. Since its only used by elements, it feels like a lot better place anyways.